Docs: no-plusplus.md - auto-semicolon insertion

xorcus opened the pull request:
The way this rule is presented in the documentation sounds as if it's just a matter of taste regarding code readability. On the other hand, due to automatic semicolon insertion, this may be considered an important rule. Of course, no one will write the code like in these examples, but some of us do not prefer whitespace affecting semantics of the source code, and that's why I believe this rule should be on by default.

Recall the famous example function which returns `undefined`:
```js
function returningUndefined() {
  return
  {
    value: 42
  }
}
```

nzakas commented on docs/rules/no-plusplus.md:
````diff
@@ -30,3 +30,26 @@ foo += 1;
 bar -= 1;
 ```
+
+## Automatic Semicolon Insertion
````
This information should be above the Rule Details section as part of the intro.

nzakas commented:
Thanks. Just one comment, we usually put intro information at the top. Feel free to update and make sure your commit message summary begins with "Docs:".

xorcus commented:
- moved the new section above Rule Details
- added Docs: prefix to the follow-up commit summary, and to the pull request summary

unfortunately, I don't know how to amend the initial commit message :) can fork all over again and make a clean commit if needed :)

nzakas commented:
We cover this information in our contributor guide: http://eslint.org/docs/developer-guide/contributing.html

You can either do an interactive rebase or you can open a new PR. Either way is fine.
